### PR TITLE
Add example for multi argument CHAR()

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v8.1 (TiDB 8.1 versions)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs/pull/16957  https://github.com/pingcap/docs/pull/16962
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -129,7 +129,7 @@ SELECT CustomerName, BIT_LENGTH(CustomerName) AS BitLengthOfName FROM Customers;
 
 ### [`CHAR()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_char)
 
-`CHAR()` 函数用于获取指定 ASCII 值的对应字符。该函数执行的操作与 `ASCII()` 相反，`ASCII()` 用于返回指定字符的 ASCII 值。
+`CHAR()` 函数用于获取指定 ASCII 值的对应字符。该函数执行的操作与 `ASCII()` 相反，`ASCII()` 用于返回指定字符的 ASCII 值。如果提供了多个参数，该函数将对所有参数执行操作，然后将它们连接在一起。
```
Comment (hfxsd) [inline functions-and-operators/string-functions.md:132]:
```suggestion
`CHAR()` 函数用于获取指定 ASCII 值的对应字符。该函数执行的操作与 `ASCII()` 相反，`ASCII()` 用于返回指定字符的 ASCII 值。如果提供了多个参数，该函数将作用于所有参数作，然后将它们连接在一起。
```

**Review by dveeden — COMMENTED** [review]
Looks good, but I'm not a good reviewer for docs-cn...

File: functions-and-operators/string-functions.md [hunk 2]
```diff
@@ -1247,22 +1260,368 @@ SELECT LPAD('TiDB',-2,'>');
...
+1 row in set (0.01 sec)
+```
+
 ### [`REGEXP_INSTR()`](https://dev.mysql.com/doc/refman/8.0/en/regexp.html#function_regexp-instr)
 
 返回满足正则的子字符串的第一个索引位置（与 MySQL 不完全兼容，具体请参考[正则函数与 MySQL 的兼容性](#正则函数与-mysql-的兼容性)）
 
+`REGEXP_INSTR(str, regexp, [start, [match, [ret, [match_type]]]])` 函数返回正则表达式（`regexp`）匹配字符串（`str`）的位置。
+
+如果 `str` 或 `regexp` 为 `NULL`，则函数返回 `NULL`。
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:1324]:
```suggestion
如果 `str` 或 `regexp` 为 `NULL`，则该函数返回 `NULL`。
```

File: functions-and-operators/string-functions.md [hunk 3]
```diff
@@ -1247,22 +1260,368 @@ SELECT LPAD('TiDB',-2,'>');
...
+```
++-----------------------------+
+| REGEXP_INSTR('abc','^.b.$') |
++-----------------------------+
+|                           1 |
++-----------------------------+
+1 row in set (0.00 sec)
+```
+
+下面示例展示了使用第三个参数来查找第二个匹配值的情况。
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:1343]:
```suggestion
下面示例展示了使用第三个参数来从字符串的指定位置起查找匹配值的情况。
```

File: functions-and-operators/string-functions.md [hunk 4]
```diff
@@ -1247,22 +1260,368 @@ SELECT LPAD('TiDB',-2,'>');
...
+```
++--------------------------------+
+| REGEXP_INSTR('abcabc','a',1,2) |
++--------------------------------+
+|                              4 |
++--------------------------------+
+1 row in set (0.00 sec)
+```
+
+下面示例展示了使用第五个参数来查找第二个匹配值后面的值的情况，而不是匹配的值。
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:1386]:
```suggestion
下面示例展示了使用第五个参数来返回匹配值后面的那个值的位置，而不是返回匹配值的位置。
```

File: functions-and-operators/string-functions.md [hunk 5]
```diff
@@ -1247,22 +1260,368 @@ SELECT LPAD('TiDB',-2,'>');
...
+|                                              0 |
++------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
 ### [`REGEXP_LIKE()`](https://dev.mysql.com/doc/refman/8.0/en/regexp.html#function_regexp-like)
 
 判断字符串是否满足正则表达式（与 MySQL 不完全兼容，具体请参考[正则函数与 MySQL 的兼容性](#正则函数与-mysql-的兼容性)）
 
+`REGEXP_LIKE(str, regex, [match_type])` 函数用于测试正则表达式是否匹配字符串。可选的 `match_type` 可以用于更改匹配行为。
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:1461]:
```suggestion
`REGEXP_LIKE(str, regex, [match_type])` 函数用于判断正则表达式是否匹配字符串。可选的 `match_type` 参数可以用于更改匹配行为。
```

File: functions-and-operators/string-functions.md [hunk 6]
```diff
@@ -1247,22 +1260,368 @@ SELECT LPAD('TiDB',-2,'>');
...
+```
++---------------------------------------+
+| REGEXP_REPLACE('TooDB', 'o', 'i',1,2) |
++---------------------------------------+
+| ToiDB                                 |
++---------------------------------------+
+1 row in set (0.00 sec)
+```
+
+下面示例中，第六个参数用于设置不区分大小写的匹配。更多关于正则表达式 `match_type` 的详细信息，请参阅 [`match_type` 兼容性](#匹配模式-match_type-兼容性)。
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:1576]:
```suggestion
下面示例中，第六个参数用于设置 `match_type` 为不区分大小写的匹配。更多关于正则表达式 `match_type` 的详细信息，请参阅 [`match_type` 兼容性](#匹配模式-match_type-兼容性)。
```

File: functions-and-operators/string-functions.md [hunk 7]
```diff
@@ -1538,6 +1897,15 @@ TiDB 与 MySQL 在 `match_type` 上的差异：
 
 - TiDB 不支持 `match_type` 为 `"u"`。
 
+| `match_type` | MySQL | TiDB | 描述                                   |
+|:------------:|-------|------|----------------------------------------|
+| c            | Yes   | Yes  | 大小写敏感匹配                          |
+| i            | Yes   | Yes  | 大小写不敏感匹配                        |
+| m            | Yes   | Yes  | Multi-line 模式                        |
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:1904]:
```suggestion
| m            | Yes   | Yes  | 匹配多行文本的模式                        |
```

File: functions-and-operators/string-functions.md [hunk 8]
```diff
@@ -1538,6 +1897,15 @@ TiDB 与 MySQL 在 `match_type` 上的差异：
...
+| s            | No    | Yes  | 匹配新行，和 MySQL 中的 `n` 相同        |
+| n            | Yes   | No   | 匹配新行，和 TiDB 中的 `s` 相同         |
+| u            | Yes   | No   | UNIX&trade，line endings               |
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:1907]:
```suggestion
| u            | Yes   | No   | UNIX&trade 换行符           |
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**